tools/proto-js-plugin/src/main/java/io/spine/js/generate/typeurl/TypeUrlMethod.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.js.generate.typeurl;
+
+import io.spine.code.proto.Type;
+import io.spine.js.generate.JsCodeGenerator;
+import io.spine.js.generate.JsOutput;
+import io.spine.js.generate.JsType;
+import io.spine.type.TypeUrl;
+
+/**
+ * Generates a method providing {@code TypeUrl} for a specific type.
+ */
+class TypeUrlMethod extends JsCodeGenerator {
+
+    private static final String METHOD_NAME = "typeUrl";
+
+    private final JsType type;
+
+    TypeUrlMethod(Type type, JsOutput jsOutput) {
+        super(jsOutput);
+        this.type = JsType.generatedFrom(type);
+    }
+
+    /**
+     * Generates the static {@code typeUrl()} method for the type.
+     */
+    @Override
+    public void generate() {
+        generateStaticMethod();
+        jsOutput().addEmptyLine();
```

Comment:
The `TypeUrlMethod` class, apparently, as well as all `JsCodeGenerator`s use an output parameter `jsOutput` instead of returning values. Such a pattern adds a side effect to the `JsCodeGenerator.generate()` method.

Instead, we should make the method [pure](https://en.wikipedia.org/wiki/Pure_function) and write the result to the output when necessary.